Templates/Common-Backend-Scripts/wazideploy-generate.sh

```diff
@@ -358,23 +458,70 @@ validateOptions() {
   if [ ! -z "${ConfigFile}" ]; then
     if [ ! -f "${ConfigFile}" ]; then
       rc=8
-      ERRMSG=$PGM": [ERROR] Specified Wazi Deploy Artifactory Configuration file (${ConfigFile}) was not found. rc="$rc
+      ERRMSG=$PGM": [ERROR] Specified Wazi Deploy Artifact repository configuration file (${ConfigFile}) was not found. rc="$rc
       echo $ERRMSG
     fi
   fi
 
   # compute the output file
   if [ -z "${PackageOutputFile}" ] && [ ! -z "${ConfigFile}" ]; then
-    # c
     checkWorkspace
     PackageOutputFile="$(wdDeployPackageDir)"
   fi
   # if relative path
-  if [[ ! ${PackageOutputFile:0:1} == "/" ]] && [[ ! -z "${PackageOutputFile}" ]] ; then
+  if [[ ! ${PackageOutputFile:0:1} == "/" ]] && [[ ! -z "${PackageOutputFile}" ]]; then
     PackageOutputFile="$(wdDeployPackageDir)/${PackageOutputFile}"
   fi
 }
 
+# When publishing is enabled, try reading the wdPackageVersionFile
+# that needs to be computed before this step.
+if [ $rc -eq 0 ] && [ "$publish" == "true" ] && [ ! -z "${buildIdentifier}" ]; then
+  checkWorkspace
+  CMD="${computePackageUrlUtil} -w $Workspace -a $App -b $Branch -i $buildIdentifier"
+
+  if [ ! -z "${PipelineType}" ]; then
+    CMD+=" -p ${PipelineType}"
+  else
+    rc=8
+    ERRMSG=$PGM": [ERROR] To compute the Package Url to automatically download the tar file via Wazi Deploy generate, you need to provide the pipelineType. rc="$rc
+    echo $ERRMSG
+  fi
+
+  if [ ! -z "${releaseIdentifier}" ]; then
+    CMD+=" -r ${releaseIdentifier}"
+  fi
+
+  if [ $rc -eq 0 ]; then
+    echo $PGM": [INFO] ** Compute Package Url based on existing conventions using command"
+    echo $PGM": [INFO] ** Invoking subtask ${CMD}"
+    ${CMD}
```

Comment:
If we remove all the unnecessary output created by the computePackaheUrlUtil script, we can retrieve the output in a variable:
```suggestion
    PackageInputFile=`${CMD}`
```
Then we don't need to go through a file to retrieve the info, it simplifies the processing.